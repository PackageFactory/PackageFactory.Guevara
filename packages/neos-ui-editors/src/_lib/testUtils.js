import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
