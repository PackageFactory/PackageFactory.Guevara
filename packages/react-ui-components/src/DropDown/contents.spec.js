import sinon from 'sinon';
import {createShallowRenderer} from './../_lib/testUtils.js';
import ShallowDropDownContents from './contents.js';

const defaultProps = {
    children: 'Foo children',
    isOpen: false,
    closeDropDown: () => null,
    theme: {
        'dropDown__contents': 'baseDropDownContentsClassName',
        'dropDown__contents--isOpen': 'openDropDownContentsClassName'
    }
};
const shallow = createShallowRenderer(ShallowDropDownContents, defaultProps);

test('should render the themes "dropDown__contents" className.', () => {
    const contents = shallow();

    expect(contents.hasClass('baseDropDownContentsClassName')).toBeTruthy();
});
test('should render the themes "dropDown__contents--isOpen" className in case the "isOpen" prop is truthy.', () => {
    const contents = shallow({
        isOpen: true
    });

    expect(contents.hasClass('openDropDownContentsClassName')).toBeTruthy();
});
test('should render the "className" prop if passed.', () => {
    const contents = shallow({
        className: 'barClassName'
    });

    expect(contents.hasClass('barClassName')).toBeTruthy();
});
test('should render a aria-hidden="true" attribute in the wrapper by default.', () => {
    const contents = shallow();

    expect(contents.html().includes('aria-hidden="true"')).toBeTruthy();
});
test('should render a aria-hidden="false" attribute in the wrapper in case the "isOpen" prop is truthy.', () => {
    const contents = shallow({
        isOpen: true
    });

    expect(contents.html().includes('aria-hidden="false"')).toBeTruthy();
});
test('should render a aria-label="dropdown" and role="button" attribute in the wrapper.', () => {
    const contents = shallow({
        isOpen: true
    });

    expect(contents.html().includes('aria-label="dropdown"')).toBeTruthy();
    expect(contents.html().includes('role="button"')).toBeTruthy();
});
test('should call the "closeDropDown" prop when clicking on the wrapper.', () => {
    const props = {
        closeDropDown: sinon.spy()
    };
    const contents = shallow(props);

    contents.simulate('click');

    expect(props.closeDropDown.calledOnce).toBeTruthy();
});
test('should render the passed children.', () => {
    const contents = shallow();

    expect(contents.html().includes('Foo children')).toBeTruthy();
});
test('should propagate the rest of the passed props to the wrapping node.', () => {
    const contents = shallow({
        id: 'fooId'
    });

    expect(contents.html().includes('id="fooId"')).toBeTruthy();
});
