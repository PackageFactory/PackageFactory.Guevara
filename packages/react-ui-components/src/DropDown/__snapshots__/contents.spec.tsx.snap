// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShallowDropDownContents/> should render correctly. 1`] = `
<ul
  aria-hidden="true"
  aria-label="dropdown"
  className="baseDropDownContentsClassName"
  onClick={[MockFunction]}
  role="listbox"
/>
`;
